lib/geomancer/geo_json.ex

```diff
@@ -15,9 +17,88 @@ defmodule Geomancer.GeoJson do
             name: nil,
             bbox: []
 
-  @spec new(features :: [Feature.t()], name :: String.t(), bbox :: map()) :: t()
-  def new(features, name, bbox) do
-    bounding = [bbox.xmin, bbox.ymin, bbox.xmax, bbox.ymax]
-    %__MODULE__{features: features, name: name, bbox: bounding}
+  @spec new({:ok, struct()} | {:error, String.t()}) :: t() | {:error, String.t()}
+  def new({:ok, %Geomancer.Shapefile{} = shapefile}) do
+    feature_set = features(shapefile.shp, shapefile.type, shapefile.dbf)
+    bbox = [shapefile.bbox.xmin, shapefile.bbox.ymin, shapefile.bbox.xmax, shapefile.bbox.ymax]
+    %__MODULE__{features: feature_set, name: shapefile.name, bbox: bbox}
   end
+
+  def new({:error, _} = error), do: error
+
+  @impl Geomancer
+  def from(input_path) do
+    case Path.extname(input_path) do
+      ".zip" ->
```

Comment:
I'm not a fan of forcing the file to have a zip extension, though I can't think of another easy way of trying to enforce the correct file being passed in. The issue is cases like we have in reaper where the file is being downloaded as a `.shapefile` and needs to be renamed before being handed off to geomancer.

Reply:
Why is it being downloaded as `.shapefile`? That's not valid at all. `.zip` is the proper extension for a Shapefile.

Reply:
I think they decided to give extensions that match the filetype which I'm not onboard with. However, I think their are other scenarios where this being a problem would be more valid (e.g. a new geo file type is created that also has .zip, or someone saves the file without an extension)

Reply:
I'm going to ignore this in this PR and follow up with another.